cpp/oneapi/dal/algo/spectral_embedding/common.cpp

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+* Copyright 2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/spectral_embedding/common.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+namespace oneapi::dal::spectral_embedding::detail {
+
+result_option_id get_embedding_id() {
+    return result_option_id{ result_option_id::make_by_index(0) };
+}
+
+template <typename Task>
+result_option_id get_default_result_options() {
+    return result_option_id{};
+}
+
+template <>
+result_option_id get_default_result_options<task::compute>() {
+    return get_embedding_id();
+}
+
+namespace v1 {
+
+template <typename Task>
+class descriptor_impl : public base {
+public:
+    explicit descriptor_impl() {}
+
+    std::int64_t embedding_dim = 0;
+    std::int64_t num_neighbors = -1;
```

Comment:
Let's align with scikit and use `components` for the dimension of projected space. Also, in oneDAL we usually use `count`, not `dim` or `num` in naming.
In DAAL it would be `numberOfComponents`, etc.
```suggestion
    std::int64_t component_count = 0;
    std::int64_t neighbor_count = -1;
```